runtime-modules/content-directory/src/lib.rs

```diff
@@ -1246,12 +1312,31 @@ decl_module! {
                 entity_controller,
             )?;
 
+            let involved_entity = value.get_involved_entity();
+
+            // Insert SingleInputPropertyValue at in_class_schema_property_id into property value vector
+            // Get VecInputPropertyValue wrapped in InputPropertyValue
+            let property_value_vector_updated = Self::insert_at_index_in_property_vector(
+                property_value_vector, index_in_property_vector, value
+            );
+
+            let class_id = entity.get_class_id();
+
+            // Ensure provided `Property` with `unique` flag set is `unique` on `Class` level
+            Self::ensure_property_unique_option_satisfied(class_id, in_class_schema_property_id, &property_value_vector_updated.simplify())?;
+
             //
             // == MUTATION SAFE ==
             //
 
+            // Insert updated property value into entity_property_values mapping at in_class_schema_property_id.
```

Comment:
Same here?

Reply:
Only default non required unique value hashes can be removed, otherwise we only update its respective hash.